sdk/typescript/src/signers/signer.ts

```diff
@@ -31,4 +36,61 @@ export abstract class Signer {
    * Returns the signature for the data and the public key of the signer
    */
   abstract signData(data: Base64DataBuffer): Promise<SignaturePubkeyPair>;
+
+  // Returns a new instance of the Signer, connected to provider.
+  // This MAY throw if changing providers is not supported.
+  abstract connect(provider: Provider): Signer;
+
+  ///////////////////
+  // Sub-classes MAY override these
+
+  /**
+   * Sign a transaction and submit to the Gateway for execution
+   *
+   * @param txBytes BCS serialised TransactionData bytes
+   */
+  async signAndExecuteTransaction(
+    txBytes: Base64DataBuffer
+  ): Promise<TransactionResponse> {
+    this._checkProvider('signAndExecuteTransaction');
+    const sig = await this.signData(txBytes);
+    return await this.provider!.executeTransaction({
+      txBytes: txBytes.toString(),
+      signature: sig.signature.toString(),
+      pubKey: sig.pubKey.toString(),
+    });
+  }
+
+  /**
+   * Serialize and Sign a `Transfer` transaction and submit to the Gateway for execution
+   */
+  async transfer(
+    transaction: TransferTransaction
+  ): Promise<TransactionResponse> {
+    this._checkProviderAndSerializer('transfer');
```

Comment:
it seems like we're always checking for prerequisite objects before doing anything, but i'm not sure we gain any significant flexibility by allowing these objects to be null/undefined at all.  

Can we set the provider, serializer, & signer in the constructor, and make them private (they're already readonly) ? that way they're always defined.
If users want to change the signer, serializer, or provider, they can construct a new class instance.

Reply:
Some use cases might just need a signer without a provider, e.g., to produce pre-signed message, to prove that the user owns this address, etc. This use case is very popular in some portfolio tracking applications such as https://debank.com/. 

My demo app also uses similar pattern: https://gist.github.com/666lcz/6cc6c983e841eda9f7a457e47bc473b7 without a provider.

Reply:
actually, now that i think about this some more - I think it seems like we should completely remove the provider from the signer. Signing & sending the tx seem like two separate concerns. 

I think the ideal signer might be something that implements _**only**_ 
```ts
interface Signer { 
    signData(data: Base64DataBuffer): Promise<SignaturePubkeyPair>
}
```